File: oop.js
// OOP

class Persona {
    nombre = ""
    apellido = ""
    edad = 0

    constructor(nombre, apellido){
        this.nombre = nombre
        this.apellido = apellido
    }

    asignarNombre(nombre){
        this.nombre = nombre
    }

    nombreCompleto(){
        return `${this.nombre} ${this.apellido}`
    }
}

let p1 = new Persona("John", "Doe") // constructor
p1.nombreCompleto() // John Doe
p1.nombre = "Jane"


class Estudiante extends Persona {
    grado = null

    constructor(nombre, apellido, grado){
        super(nombre, apellido)
        this.grado = grado
    }
}

let est1 = new Estudiante("John", "Doe", "1ero")
console.log(est1.nombreCompleto())

File: oop.py
# OOP

class Persona:
    nombre = ""
    apellido = ""
    edad = 0

    def __init__(self, nombre, apellido):
        self.nombre = nombre
        self.apellido = apellido

    def nombre_completo(self):
        return f"{self.nombre} {self.apellido}"
    

p1 = Persona("Jane", "Doe")
print(p1.nombre_completo())

class Estudiante(Persona):
    grado = None

    def __init__(self, nombre, apellido, grado):
        super().__init__(nombre, apellido)
        self.grado = grado

    def nombre_completo(self):
        return f"Estudiante: {self.nombre} {self.apellido}"
    

est1 = Estudiante("Jane", "Doe", "3ero")
print(est1.nombre_completo())


def sumar(a, b):
    return a + b